src/app/main/main.component.ts

```diff
@@ -162,10 +182,36 @@ export class MainComponent implements OnInit, OnDestroy {
     StyleProps.forEach((prop) => stylesGroupMap[prop.name] = this.createStyleFormGroup());
     this.stylesFormGroup = this._formBuilder.group(stylesGroupMap);
 
+    // Sharing form group
+    this.sharingFormGroup = this._formBuilder.group({
+      sharingUrl: '',
+    });
+
     // Initialize default styles.
     this.updateStyles();
   }
 
+  saveDataToSharedStorage() {
+    const dataValues = this.dataFormGroup.getRawValue();
+    // Encrypt the style values using the sql string.
+    const hashedStyleValues = CryptoJS.AES.encrypt(JSON.stringify(this.styles), this.jobWrappedSql + this.bytesProcessed);
+    const shareableData = {
+      sharingVersion: SHARING_VERSION,
+      projectID: dataValues.projectID,
+      jobID: this.jobID,
+      location: dataValues.location,
+      styles: hashedStyleValues.toString(),
+      creation_timestamp_ms: Date.now()
```

Comment:
any reason for the `camel_case` here? Is this a firebase thing?

Reply:
Done!